harness/determined/searcher/core_search_runner.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import os
+import pickle
+import time
+from typing import Any, Dict, List, Optional, Tuple
+
+import determined as det
+from determined.common.api.bindings import (
+    determinedexperimentv1State,
+    get_GetExperiment,
+    post_PauseExperiment,
+)
+from determined.experimental import client
+from determined.searcher.search_method import Operation, SearchMethod
+from determined.searcher.search_runner import SearchRunner
+
+
+class CoreSearchRunner(SearchRunner):
+    def __init__(self, search_method: SearchMethod, context: det.core.Context) -> None:
+        super().__init__(search_method)
+        self.context = context
+        self.info = det.get_cluster_info()
+
+        assert self.info is not None, "CoreSearchRunner only runs on-cluster"
+        self.latest_checkpoint = self.info.latest_checkpoint
+
+    def run(
+        self,
+        exp_config: Dict[str, Any],
+        context_dir: Optional[str] = None,
+    ) -> int:
+        logging.info("CoreSearchRunner.run")
```

Comment:
probably worth using a logger named `det.searcher` for anything in this module.

Eventually we'll move the whole codebase to look like that.